recipe/meta.yaml

```
@@ -69,11 +62,10 @@ outputs:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - bazel <=3.4.1  # [not win]
-        - bazel <3.7     # [win]
+        - bazel =4.2.1
```

Before pouring too much time into bazel, I'd suggest to first try reverting this.
```suggestion
        - bazel <=3.4.1  # [not win]
        - bazel <3.7     # [win]
```